tortoise/full.go

```diff
@@ -60,14 +64,32 @@ func (f *full) countBallot(logger log.Log, ballot *ballotInfo) {
 	}
 }
 
+func (f *full) countForLateBlock(block *blockInfo) {
+	// ballots are always added to the state after blocks that are
+	// explicitly referenced in the ballot.
```

Comment:
this many not be true in the near future with delayed fetching, right?

Reply:
in case of delayed fetching - block identifier will be added together with a first ballot that references it

Reply:
right. the struct that contains block id and tick count.